### PR TITLE
adds initial list

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/4225.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,6 +151,22 @@ just keep in mind these considerations:
 * The kube config file path for connecting to the test cluster can be obtained from
 the `DefaultKubeConfig` constant in the `giantswarm/e2e-harness/pkg/harness` package.
 
+## Projects using `e2e-harness`
+
+- https://github.com/giantswarm/apprclient
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply README.md]: I think this is good as a start, we need to kind of have a pointer when someone does a breaking change where it needs to be vendored.

**Review by MarcelMue — APPROVED** [review]
LGTM, Thanks <3

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kopiczko) [reply README.md]: That's very good idea. We should maintain such list in general.
I think project from there are missing https://github.com/giantswarm/giantswarm/issues/4076.

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply README.md]: Thanks. I added the missing ones from the other issue's list.

**Pull request merged**